ddtrace/contrib/internal/langchain/patch.py

```diff
@@ -974,6 +975,112 @@ def traced_similarity_search(langchain, pin, func, instance, args, kwargs):
     return documents
 
 
+@with_traced_module
+def traced_base_tool_invoke(langchain, pin, func, instance, args, kwargs):
+    integration = langchain._datadog_integration
+    tool_input = get_argument_value(args, kwargs, 0, "input")
+    config = get_argument_value(args, kwargs, 1, "config", optional=True)
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s.%s" % (func.__module__, func.__class__.__name__, func.__name__, func.__self__.name),
+        interface_type="tool",
+        submit_to_llmobs=True,
+    )
+
+    tool_output = None
+    try:
+        tool_attributes = ["name", "description", "metadata", "tags"]
+
+        for attribute in tool_attributes:
+            value = getattr(instance, attribute, None)
+            if isinstance(value, dict):
+                for key, meta_value in value.items():
+                    span.set_tag_str("langchain.request.tool.%s.%s" % (attribute, key), str(value))
+            elif isinstance(value, list):
+                for idx, tag in enumerate(value):
+                    span.set_tag_str("langchain.request.tool.%s.%d" % (attribute, idx), str(value))
+            elif value is not None:
+                span.set_tag_str("langchain.request.tool.%s" % attribute, str(value))
+
+        if integration.is_pc_sampled_span(span):
+            if tool_input:
+                span.set_tag_str("langchain.request.input", integration.trunc(str(tool_input)))
+            if config:
+                span.set_tag_str("langchain.request.config", json.dumps(config))
+        tool_output = func(*args, **kwargs)
+        if tool_output is not None:
+            if integration.is_pc_sampled_span(span):
+                span.set_tag_str("langchain.response.output", integration.trunc(str(tool_output)))
+    except Exception:
+        span.set_exc_info(*sys.exc_info())
+        raise
+    finally:
+        if integration.is_pc_sampled_llmobs(span):
+            integration.llmobs_set_tags(
+                "tool",
+                span,
+                tool_input,
```

Comment:
Is it possible to also pass in the `config` object into this so we can potentially tag metadata/tags specified in the tool invocation?

Reply:
I'll pass an object containing the tool_input, config, and tool_info (tags, metadata, name, description)